frontend/src/components/pages/RegistrantExperience/index.tsx

```diff
@@ -42,49 +73,75 @@ const RegistrantExperiencePage = (): React.ReactElement => {
     name: "",
     waiverCompleted: false,
   });
-  useEffect(() => {
-    CampsAPIClient.getCampById(campId).then((campResponse) => {
-      if (campResponse.id) {
-        setCamp(campResponse);
-      }
-      setIsLoading(false);
-    });
-
-    AdminAPIClient.getWaiver().then((waiver) => {
-      waiverDispatch({
-        type: WaiverActions.LOADED_WAIVER,
-        waiver,
-      });
-    });
-  }, [campId]);
 
   const [samplePersonalInfo, setSamplePersonalInfo] = useState(false);
   const [sampleAdditionalInfo, setSampleAdditionalInfo] = useState(false);
-  const [sampleRegisterField, setSampleRegisterField] = useState(false);
 
   const isPersonalInfoFilled = samplePersonalInfo;
   const isAdditionalInfoFilled = sampleAdditionalInfo;
   const isWaiverFilled = waiverInterface.waiverCompleted;
-  const isReviewRegistrationFilled = sampleRegisterField;
+  const isReviewRegistrationFilled = true;
 
-  const isCurrentStepCompleted = (step: RegistrantExperienceSteps) => {
-    switch (step) {
-      case RegistrantExperienceSteps.PersonalInfoPage:
-        return isPersonalInfoFilled;
-      case RegistrantExperienceSteps.AdditionalInfoPage:
-        return isAdditionalInfoFilled;
-      case RegistrantExperienceSteps.WaiverPage:
-        return isWaiverFilled;
-      case RegistrantExperienceSteps.ReviewRegistrationPage:
-        return isReviewRegistrationFilled;
-      default:
-        return false;
+  const [registeringCampers, setRegisteringCampers] = useState<
+    CreateCamperRequest[]
+  >(dummyCampers);
+
+  const [errorModalMessage, setErrorModalMessage] = useState<ErrorModalMessage>(
+    {
+      title: "",
+      body: "",
+    },
+  );
+
+  const saveSessionToSessionStorage = (
+    sessionCampId: string, // better named `campId`, but linter doesn't like
+    checkoutData: CheckoutData,
+  ) => {
+    sessionStorage.setItem(CAMP_ID_SESSION_STORAGE_KEY, sessionCampId);
+    sessionStorage.setItem(
+      getCheckoutSessionStorageKey(sessionCampId),
+      JSON.stringify(checkoutData),
+    );
+  };
+
+  const goToCheckout = (
+    checkoutSessionUrl: string,
+    currentCamp: CampResponse,
```

Comment:
ehh it's a stylistic thing that I've never fully resolved, whether methods should stay isolated from state dependencies and pass through different functions, especially since they can instead exist out of component. I guess in this case it reduces boilerplate by using state vars?